File: Java/Fundamentals/Language/OOP/Interval.java
class Interval{

	private int min, sec;

	public Interval(int m, int s){
		min = m + s / 60;
		sec = s % 60;
	}

	public int minutes(){return min;}

	public int seconds(){return sec;}

	public int time(){
		return 60 * min + sec;
	}

	public String toString(){
		return String.format("%d:%2d", min, sec);
	}

	public int hashCode(){
		return 1000 * time();
	}

	public boolean equals(Object other){
		if(other instanceof Interval){
			Interval that = (Interval)other;
			return (this.min == that.min) && (this.sec == that.sec);
		}
		return false;
	}

	public Interval add(Interval other){
		return new Interval(min + other.min, sec + other.sec);
	}
}

File: MS.NET/Fundamentals/Generic/Basics/interval.cs
using System;

partial class Interval : IComparable<Interval>
{
	private int min;
	private int sec;

	partial void OnCreate();

	public Interval(int m, int s)
	{
		min = m + s / 60;
		sec = s % 60;
		OnCreate();
	}

	public int Minutes => min;

	public int Seconds => sec;

	public int Time
	{
		get
		{
			return 60 * min + sec;
		}
	}

	public override string ToString()
	{
		return $"{min}:{sec:00}"; //string.Format("{0}:{1:00}", min, sec)
	}

	public override int GetHashCode()
	{
		return 1000 * Time;
	}

	public override bool Equals(object other)
	{
		if(other is Interval)
		{
			Interval that = (Interval)other;
			return (this.min == that.min) && (this.sec == that.sec);
		}
		
		return false;
	}

	public int CompareTo(Interval that)
	{
		return this.Time - that.Time;
	}
}

